rtl/eclrtl/eclregex.cpp

```diff
@@ -286,17 +319,18 @@ class CCompiledStrRegExpr : implements ICompiledStrRegExpr
             {
                 PCRE2_SIZE * ovector = pcre2_get_ovector_pointer_8(matchData);
                 const char * matchStart = _subject + ovector[0];
-                unsigned matchLen = ovector[1] - ovector[0];
+                unsigned matchSize = ovector[1] - ovector[0];
 
-                // Copy match to output buffer
-                out.ensureAvailable(outBytes + matchLen + sizeof(size32_t));
+                // Copy match to output buffer; this is number of characters used
+                // by the string, not the number of bytes
+                out.ensureAvailable(outBytes + matchSize + sizeof(size32_t));
                 byte * outData = out.getbytes() + outBytes;
-                * (size32_t *) outData = matchLen;
-                memcpy(outData + sizeof(size32_t), matchStart, matchLen);
-                outBytes += matchLen + sizeof(size32_t);
+                * (size32_t *) outData = (isUTF8Enabled ? rtlUtf8Length(matchSize, matchStart) : matchSize);
+                memcpy(outData + sizeof(size32_t), matchStart, matchSize);
+                outBytes += matchSize + sizeof(size32_t);
 
                 // Update offset
-                offset += matchLen + 1;
+                offset += matchSize + 1;
```

Comment:
Not new code, but is this +1 correct?   E.g. it suggests that matching "ab" against "abab" would only get the first match.

Reply:
Better yet:

offset = ovector[1];